app/validation.go

```diff
@@ -130,13 +127,19 @@ func collectManifestProperties(roleManifest *model.RoleManifest) map[string]stri
 
 	// Per-instance-group properties
 	for _, instanceGroup := range roleManifest.InstanceGroups {
-		for property, template := range instanceGroup.Configuration.Templates {
-			properties[property] = template
+		for _, propertyDef := range instanceGroup.Configuration.Templates {
+			property := propertyDef.Key.(string)
+			template := propertyDef.Value
+
+			properties[property] = fmt.Sprintf("%v", template)
 		}
 	}
 
 	// And the global properties
-	for property, template := range roleManifest.Configuration.Templates {
+	for _, propertyDef := range roleManifest.Configuration.Templates {
+		property := propertyDef.Key.(string)
+		template := propertyDef.Value.(string)
```

Comment:
This is always castable right?

Reply:
Globals yes.
They are not allowed to be constants - they need to contain a string.